Fix woocommerce admin storybook commands (post WooCommerce Admin merge)

### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

Running `pnpm nx storybook woocommerce-admin` failed 
```
ModuleBuildError: Module build failed (from ../../node_modules/.pnpm/babel-loader@8.2.3_b72fb7e629d39881e138edb6dcd0dfbe/node_modules/babel-loader/lib/index.js):
Error: /Users/chihsuan/Projects/woocommerce/plugins/woocommerce-admin/babel.config.js: Error while loading config - Cannot find module '/Users/chihsuan/Projects/woocommerce/packages/js/e2e-environment/node_modules/@woocommerce/api/dist/index.js'. Please verify that the package.json has a valid "main" entry
    at tryPackage (node:internal/modules/cjs/loader:353:19)
    at Function.Module._findPath (node:internal/modules/cjs/loader:566:18)
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:919:27)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/Users/chihsuan/Projects/woocommerce/packages/js/e2e-environment/utils/update-ready-page.js:2:31)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1157:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/Users/chihsuan/Projects/woocommerce/packages/js/e2e-environment/utils/index.js:10:31)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1157:10)
    at processResult (/Users/chihsuan/Projects/woocommerce/node_modules/.pnpm/webpack@5.70.0_webpack-cli@4.9.2/node_modules/webpack/lib/NormalModule.js:758:19)
    at /Users/chihsuan/Projects/woocommerce/node_modules/.pnpm/webpack@5.70.0_webpack-cli@4.9.2/node_modules/webpack/lib/NormalModule.js:860:5
    at /Users/chihsuan/Projects/woocommerce/node_modules/.pnpm/loader-runner@4.2.0/node_modules/loader-runner/lib/LoaderRunner.js:399:11
    at /Users/chihsuan/Projects/woocommerce/node_modules/.pnpm/loader-runner@4.2.0/node_modules/loader-runner/lib/LoaderRunner.js:251:18
    at context.callback (/Users/chihsuan/Projects/woocommerce/node_modules/.pnpm/loader-runner@4.2.0/node_modules/loader-runner/lib/LoaderRunner.js:124:13)
    at /Users/chihsuan/Projects/woocommerce/node_modules/.pnpm/babel-loader@8.2.3_b72fb7e629d39881e138edb6dcd0dfbe/node_modules/babel-loader/lib/index.js:59:103
ModuleBuildError: Module build failed (from ../../node_modules/.pnpm/babel-
```

### Expected behavior

Can build stories and start the storybook server

### Actual behavior

Failed to build stories.

### Steps to reproduce

`pnpm nx storybook woocommerce-admin`

### WordPress Environment

N/A

### Isolating the problem

- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

--- a/packages/js/components/package.json
+++ b/packages/js/components/package.json
@@ -77,6 +77,7 @@
 	},
 	"devDependencies": {
 		"@babel/core": "^7.17.5",
+		"@babel/runtime": "^7.17.2",
 		"@storybook/addon-actions": "^6.4.0",
 		"@storybook/addon-console": "^1.2.3",
 		"@storybook/addon-controls": "^6.4.19",

--- a/packages/js/data/package.json
+++ b/packages/js/data/package.json
@@ -43,6 +43,7 @@
 	},
 	"devDependencies": {
 		"@babel/core": "^7.17.5",
+		"@babel/runtime": "^7.17.2",
 		"@testing-library/react": "^12.1.3",
 		"@testing-library/react-hooks": "^7.0.2",
 		"@types/wordpress__data-controls": "^2.2.0",

--- a/packages/js/navigation/package.json
+++ b/packages/js/navigation/package.json
@@ -51,6 +51,7 @@
 		"@babel/core": "^7.17.5",
 		"@wordpress/eslint-plugin": "^11.0.0",
 		"eslint": "^8.12.0",
+		"@babel/runtime": "^7.17.2",
 		"jest": "^27.5.1",
 		"jest-cli": "^27.5.1",
 		"rimraf": "^3.0.2",

--- a/packages/js/number/package.json
+++ b/packages/js/number/package.json
@@ -39,6 +39,7 @@
 		"@babel/core": "^7.17.5",
 		"@wordpress/eslint-plugin": "^11.0.0",
 		"eslint": "^8.12.0",
+		"@babel/runtime": "^7.17.2",
 		"jest": "^27.5.1",
 		"jest-cli": "^27.5.1",
 		"rimraf": "^3.0.2",

--- a/plugins/woocommerce-admin/package.json
+++ b/plugins/woocommerce-admin/package.json
@@ -235,6 +235,7 @@
 		"eslint-import-resolver-webpack": "^0.13.2",
 		"eslint-plugin-import": "^2.25.4",
 		"eslint-plugin-react": "^7.29.2",
+		"expose-loader": "^3.1.0",
 		"fork-ts-checker-webpack-plugin": "^6.5.0",
 		"fs-extra": "^8.1.0",
 		"grunt": "^1.4.1",

--- a/plugins/woocommerce-admin/storybook/.storybook/main.js
+++ b/plugins/woocommerce-admin/storybook/.storybook/main.js
@@ -6,9 +6,9 @@ module.exports = {
 	},
 	stories: [
 		// WooCommerce Admin / @woocommerce/components components
-		'../../packages/components/src/**/stories/*.@(js|tsx)',
+		'../../../../packages/js/components/src/**/stories/*.@(js|tsx)',
 		// WooCommerce Admin / @woocommerce/experimental components
-		'../../packages/experimental/src/**/stories/*.@(js|tsx)',
+		'../../../../packages/js/experimental/src/**/stories/*.@(js|tsx)',
 		'../../client/**/stories/*.js',
 	],
 	addons: [

--- a/plugins/woocommerce-admin/storybook/webpack.config.js
+++ b/plugins/woocommerce-admin/storybook/webpack.config.js
@@ -22,14 +22,32 @@ const wcAdminPackages = [
 ];
 
 module.exports = ( storybookConfig ) => {
-	storybookConfig.module.rules.push( ...wcAdminWebpackConfig.module.rules );
+	storybookConfig.module.rules = [
+		...storybookConfig.module.rules,
+		...wcAdminWebpackConfig.module.rules,
+		// We need to expose packages in "peerDependencies" to the global scope for @woocommerce/* to resolve packages.
+		{
+			test: require.resolve( 'moment' ),
+			loader: 'expose-loader',
+			options: {
+				exposes: [ 'moment' ],
+			},
+		},
+		{
+			test: require.resolve( '@wordpress/data' ),
+			loader: 'expose-loader',
+			options: {
+				exposes: [ '_wp_data' ],
+			},
+		},
+	];
 
 	storybookConfig.resolve.alias = wcAdminWebpackConfig.resolve.alias;
 
 	wcAdminPackages.forEach( ( name ) => {
 		storybookConfig.resolve.alias[
 			`@woocommerce/${ name }`
-		] = path.resolve( __dirname, `../packages/${ name }/src` );
+		] = path.resolve( __dirname, `../../../packages/js/${ name }/src` );
 	} );
 
 	storybookConfig.resolve.alias[ '@woocommerce/settings' ] = path.resolve(
@@ -53,20 +71,25 @@ module.exports = ( storybookConfig ) => {
 				{
 					from: path.resolve(
 						__dirname,
-						`../packages/components/build-style/*.css`
+						`../../../packages/js/components/build-style/*.css`
 					),
 					to: `./component-css/[name][ext]`,
 				},
 				{
 					from: path.resolve(
 						__dirname,
-						`../packages/experimental/build-style/*.css`
+						`../../../packages/js/experimental/build-style/*.css`
 					),
 					to: `./experimental-css/[name][ext]`,
 				},
 			],
 		} )
 	);
+
+	storybookConfig.externals = {
+		'@wordpress/data': '_wp_data',
+		moment: 'moment',
+	};
 
 	return storybookConfig;
 };

